bitcoin/chainparams.c

```diff
@@ -123,3 +124,20 @@ const struct chainparams *chainparams_by_bip173(const char *bip173_name)
 	}
 	return NULL;
 }
+
+const char *chainparams_networks(void)
+{
+    static char* nwstr = NULL;
+    if (!nwstr) {
+        size_t len = (ARRAY_SIZE(networks) << 1) - 1; // 1 too many ", " counted -> 1 byte extra for final \0
+        for (size_t i = 0; i < ARRAY_SIZE(networks); ++i) {
+            len += strlen(networks[i].network_name);
+        }
+        nwstr = malloc(len);
+        char* ptr = nwstr + sprintf(nwstr, "%s", networks[0].network_name);
+        for (size_t i = 1; i < ARRAY_SIZE(networks); ++i) {
+            ptr += sprintf(ptr, ", %s", networks[i].network_name);
```

Comment:
Use `tal_fmt` from https://github.com/ElementsProject/lightning/blob/master/ccan/ccan/tal/str/str.h instead.  `#include<ccan/tal/str/str.h>`.

https://github.com/ElementsProject/lightning/blob/5179025977b1179ba78f5c1c27cc95c99eb17543/ccan/ccan/tal/str/str.h#L35-L45

Have `chainparams_networks` accept a `const tal_t *ctx` argument, which is what you pass as first argument to `tal_fmt`.

Use `tal_append_fmt` to append.

So something like:

    networks = tal_fmt(ctx, "%s", networks[0].network_name);
    for (i = 0; i < ARRAY_SIZE(networks); ++i)
        networks = tal_append_fmt(ctx, take(networks), ", %s", networks[i].network_name);
    return networks;

Don't bother with keeping a `static` var, just always allocate a new string, this is not performance-critical code like signing or routefinding.

Reply:
Nitpick: `s/chainparams_networks/chainparams_get_network_names/` maybe?

Reply:
Updated the code to use `tal_t`. (Neat library!)